src/ClientOptionsTrait.php

```diff
@@ -261,6 +261,10 @@ private static function determineMtlsEndpoint(string $apiEndpoint)
     private function createCredentialsWrapper($credentials, array $credentialsConfig, string $universeDomain)
     {
         if (is_null($credentials)) {
+            // If the user has explicitly set the apiKey option, use Api Key credentials
+            if (isset($credentialsConfig['apiKey'])) {
+                return ApiKeyHeaderCredentials::build($credentialsConfig);
+            }
```

Comment:
I want to call out these two conditions from the API Key Design Doc.

1. If both credentials and api key are explicitly set, return an error to the user.
2. If api key is set, do not attempt the ADC flow.

I think the first condition is missing in this code. 

Is the second condition also missing?

Reply:
In the block you've highlighted, the logic goes like this:

 -  _IF_ `$credentials` are null (e.g. credentials have not been explicitly set), _THEN_
     - _IF_ the `apiKey` option has been set, _THEN_ create `ApiKeyHeaderCredentials`
     - _ELSE_` use `applicationDefaultCredentials` (this happens in `CredentialsWrapper::build`)

So no, the second condition is not missing.

I can add validation for the first at the top of this function.